### PR TITLE
Add truffle v5 target, switch from BigNumber to BN

### PR DESCRIPTION
**Pull request by pash7ka**
As mentioned in issue #193 and pull requst #184, with v5.0.0 truffle [moved](https://github.com/trufflesuite/truffle/releases/tag/v5.0.0) from [Web3.js](https://github.com/ethereum/web3.js/) v0.x to v1.x. One of major change in it: use of [BN.js](https://github.com/indutny/bn.js/) instead of [bignumber.js](https://mikemcl.github.io/bignumber.js/).
With this pull request i've introduced `typechain-target-truffle-v5` based on @MicahZoltu pr #184.

### CODE REVIEW

File: packages/typechain-target-truffle-v5/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Typechain target Truffle
```
Comment (krzkaczor) [inline packages/typechain-target-truffle-v5/README.md]: can u add here `Truffle@5` or something?

Comment (pash7ka) [reply packages/typechain-target-truffle-v5/README.md]: done

**Comment by pash7ka** [comment]
> Isn't truffle@5 using web3@1 under the hood? Does it mean it has the same api? Can we somehow reuse the generated types between packages?
Yes, they use web3@1 under the hood, butthey created their own wrappers around it.
For example, when [deploying a contract](https://github.com/trufflesuite/truffle/blob/v5.1.4/packages/contract/lib/execute.js#L285) they wrap it with [TruffleContract](https://github.com/trufflesuite/truffle/blob/v5.1.4/packages/contract/lib/contract/constructorMethods.js). Also they [wrap receipt](https://github.com/trufflesuite/truffle/blob/v5.1.4/packages/contract/lib/handlers.js#L132) of a call.

**Comment by krzkaczor** [comment]
@pash7ka oh, makes sense. I will finish CR today 👍

**Comment by krzkaczor** [comment]
Merged to tmp branch for now - it needs a little bit more work. 

Thanks for this @pash7ka @MicahZoltu ! 🎉

**Pull request merged**